core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -858,19 +844,33 @@ object SparkSubmit extends CommandLineUtils {
     require(path != null, "path cannot be null.")
     val uri = Utils.resolveURI(path)
     uri.getScheme match {
-      case "file" | "local" =>
-        path
-
+      case "file" | "local" => path
+      case "https" | "http" | "ftp" => path
```

Comment:
My original thinking is that `FileSystem` API doesn't support such API, so I treat them separately without downloading. Will verify it.